1-ES6/app.js

```diff
@@ -0,0 +1,46 @@
+class App {
+	constructor() {
+		this.rootEl = document.querySelector('.root');
+		this.requestService = new RequestService();
+		this.sidebar = new Sidebar();
+		this.pageContent = new PageContent();
+		this.sourcesList = new SourcesList();
+		this.newsList = new NewsList();
+	}
+
+	initEventHandlers() {
+		this.sourcesList.initCheckboxHandler(this.getEventHandlersCallbacks());
+	}
+
+	buildLayout([{
+		articles
+	}, {
+		sources
+	}]) {
+		const newsListComponent = this.newsList.build(articles).getComponent();
+		const pageContentComponent = this.pageContent.build(newsListComponent).getComponent();
+		const sourcesListComponent = this.sourcesList.build(sources).getComponent();
+		const sidebarComponent = this.sidebar.build(sourcesListComponent).getComponent();
+
+		this.rootEl.appendChild(sidebarComponent);
```

Comment:
It would be better to create a fragment and build all the stuff in it and append this fragment to root at once

Reply:
Done. Thanks!